### PR TITLE
DT-601 changes to support retrieval of references codes in relation court hearing creation.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This change mainly includes support for retrieval of reference codes (and associated refactoring) so it can be used during the creation of court hearings.

### CODE REVIEW

File: src/main/java/net/syscon/elite/service/CourtHearingsService.java [hunk 1]
```diff
@@ -27,35 +27,43 @@
...
         // TODO throw entity not found exception
         var offenderBooking = offenderBookingRepository.findById(bookingId).orElseThrow();
 
-        // TODO throw entity not found exception
+        // TODO throw entity not found exception (check is active)
         var courtCase = offenderBooking.getCourtCaseBy(hearing.getCourtCaseId()).orElseThrow();
 
         // TODO throw entity not found exception (check is prison?)
-        AgencyLocation fromPrison = agencyLocationRepository.findById(hearing.getFromPrisonLocation()).orElseThrow();
+        var locationOfPrison = agencyLocationRepository.findById(hearing.getFromPrisonLocation()).orElseThrow();
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/service/CourtHearingsService.java]: final

File: src/main/java/net/syscon/elite/service/CourtHearingsService.java [hunk 2]
```diff
@@ -27,35 +27,43 @@
...
 
         // TODO throw entity not found exception (check is court)
-        AgencyLocation toCourt = agencyLocationRepository.findById(hearing.getToCourtLocation()).orElseThrow();
+        var locationOfCourt = agencyLocationRepository.findById(hearing.getToCourtLocation()).orElseThrow();
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/service/CourtHearingsService.java]: final

File: src/main/java/net/syscon/elite/service/CourtHearingsService.java [hunk 3]
```diff
@@ -27,35 +27,43 @@
...
 
         // TODO sort out reference codes!!!
         CourtEvent courtEvent = CourtEvent.builder()
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/service/CourtHearingsService.java]: final var

File: src/main/java/net/syscon/elite/repository/jpa/model/ReferenceCode.java [hunk 4]
```diff
@@ -16,16 +18,26 @@
...
     private String domain;
+
     @Id
     private String code;
 
     private String description;
 
-    static final String CASE_STS = "CASE_STS";
-    static final String LEG_CASE_TYP = "LEG_CASE_TYP";
     static final String MLTY_BRANCH = "MLTY_BRANCH";
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/repository/jpa/model/ReferenceCode.java]: should these still be here

Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/repository/jpa/model/ReferenceCode.java]: I really wanted to keep the changes specific to court stuff.  Was planning on do this in a separate PR.

**Pull request merged**